### PR TITLE
[cxx-interop] Fix Constexpr issue

### PR DESCRIPTION
**Pull request by Huddie**
Working on fixing constexpr on windows. Using swift-ci to test

### CODE REVIEW

**Comment by Huddie** [comment]
@swift-ci please test Windows platform

**Comment by Huddie** [comment]
@swift-ci please test Windows platform

File: lib/ClangImporter/ImportDecl.cpp [hunk 1]
```diff
@@ -3209,11 +3209,21 @@ namespace {
     AccessorDecl *tryCreateConstexprAccessor(const clang::VarDecl *clangVar,
                                              VarDecl *swiftVar) {
       assert(clangVar->isConstexpr());
+
+      // Taken from clang/AST/Decl.cpp
+      // We must check if this eval is not value dependent
+      // as the assert there will fail otherwise
+      clang::EvaluatedStmt *Eval = clangVar->ensureEvaluatedStmt();
```
Comment (zoecarver) [inline lib/ClangImporter/ImportDecl.cpp:3216]:
```suggestion
      clang::EvaluatedStmt *eval = clangVar->ensureEvaluatedStmt();
```

File: lib/ClangImporter/ImportDecl.cpp [hunk 2]
```diff
@@ -3209,11 +3209,21 @@ namespace {
...
+      const auto *Init = cast<clang::Expr>(Eval->Value);
```
Comment (zoecarver) [inline lib/ClangImporter/ImportDecl.cpp:3217]:
```suggestion
      const auto *init = cast<clang::Expr>(Eval->Value);
```

File: lib/ClangImporter/ImportDecl.cpp [hunk 3]
```diff
@@ -3209,11 +3209,21 @@ namespace {
...
+
+      // Bail if the assert in clang/AST/Decl will fail
```
Comment (zoecarver) [inline lib/ClangImporter/ImportDecl.cpp:3219]: This isn't really a helpful comment. Maybe we can figure out why dependent values don't work? Or just state that dependent values don't work.

File: test/Interop/Cxx/stdlib/msvcprt-module-interface.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
+// RUN: %empty-directory(%t)
```
Comment (zoecarver) [inline test/Interop/Cxx/stdlib/msvcprt-module-interface.swift:1]: Is this the test from @compnerd's patch?

**Comment by zoecarver** [comment]
Looks good to me sans the nits.

File: test/Interop/Cxx/stdlib/msvcprt-module-interface.swift (hunk 4 above)
Comment (Huddie) [reply test/Interop/Cxx/stdlib/msvcprt-module-interface.swift:1]: Yes just using for testing on ci. I can drop it

**Pull request closed without merging**